core/Pos/Util/Util.hs

```diff
@@ -190,6 +196,33 @@ instance ToJSON Byte where
 instance IsString s => MonadFail (Either s) where
     fail = Left . fromString
 
+instance Rand.DRG drg => HasLoggerName (Rand.MonadPseudoRandom drg) where
+    getLoggerName = pure mempty
+    modifyLoggerName = flip const
+
+#define MONADRANDOM(CTX, MONAD) \
+    instance CTX => Rand.MonadRandom (MONAD) where \
+        getRandomBytes = lift . Rand.getRandomBytes \
+
+MONADRANDOM( Rand.MonadRandom m            , ReaderT r m)
+MONADRANDOM((Rand.MonadRandom m, Monoid w) , Lazy.WriterT w m)
+MONADRANDOM((Rand.MonadRandom m, Monoid w) , Strict.WriterT w m)
+MONADRANDOM( Rand.MonadRandom m            , Lazy.StateT s m)
+MONADRANDOM( Rand.MonadRandom m            , Strict.StateT s m)
+MONADRANDOM((Rand.MonadRandom m, Monoid w) , Lazy.RWST r w s m)
+MONADRANDOM((Rand.MonadRandom m, Monoid w) , Strict.RWST r w s m)
+MONADRANDOM( Rand.MonadRandom m            , ExceptT e m)
+MONADRANDOM( Rand.MonadRandom m            , NamedPureLogger m)
+
+instance (Monad m, Monad (t m), MonadTrans t, Rand.MonadRandom m)
+         => Rand.MonadRandom (Ether.TaggedTrans tag t m) where
+    getRandomBytes = lift . Rand.getRandomBytes
```

Comment:
1. Why didn't you use the `MONADRANDOM` macro here?
2. Why not define overlappable instance for `MonadTrans`?